tests/optimizations/run_all.py

```diff
@@ -108,12 +108,15 @@ def checkSequence(statements):
             print_arg, = getRole(statement, "value")
 
             if not isConstantExpression(print_arg):
-                sys.exit(
-                    "%s: Error, print of non-constant '%s'." % (
-                        getSourceRef(statement),
-                        getKind(print_arg)
+                if isinstance(search_mode, SearchModeAll):
```

Comment:
One style complaint here. I think using `isinstance` is nearly always too involved with classes and their hierarchy. Instead, most often, a `isXYZ` or `shallABC` method that is asked for a feature or desired behavior better.

Also, in this concrete example, I dislike the `sys.exit()` being conditional. I think that is precisely what we want the search mode to decide.

Instead call `search_mode.exit(filename, message =...)`, the default implementation will then call `sys.exit(message)` whereas the all mode can pick it up, and add it to the list of errors occurred.